sockeye/inference.py

```diff
@@ -925,19 +975,82 @@ def get(self, lengths: Union[mx.nd.NDArray, int, float]) -> Union[mx.nd.NDArray,
         return self.hybrid_forward(None, lengths)
 
 
+def _concat_nbest_translations(translations: List[Translation], stop_ids: Set[int],
+                               length_penalty: LengthPenalty) -> Translation:
+    """
+    Combine nbest translations through concatenation.
+
+    :param translations: A list of translations (sequence starting with BOS symbol,
+        attention_matrix), score and length.
+    :param translations: The EOS symbols.
+    :return: A concatenation of the translations with a score.
+    """
+    expanded_translations = [_expand_nbest_translation(translation) for translation in translations]
```

Comment:
How about a generator:
```suggestion
    expanded_translations = (_expand_nbest_translation(translation) for translation in translations)
```